### PR TITLE
[wasm][debugger] Force sourceUrl with a suffix .cdp

### PR DESCRIPTION
**Pull request by thaystg**
Add .cdp suffix to sourceUrl to all the evaluation that is done internally by the dotnet debugger.
This will avoid slowness in VS.

### CODE REVIEW

File: src/mono/browser/debugger/BrowserDebugProxy/DevToolsHelper.cs [hunk 1]
```diff
@@ -324,6 +324,14 @@ public static MonoCommands SendDebuggerAgentCommandWithParms(int runtimeId, int
         public static MonoCommands ReleaseObject(int runtimeId, DotnetObjectId objectId) => new MonoCommands($"getDotnetRuntime({runtimeId}).INTERNAL.mono_wasm_release_object('{objectId}')");
 
         public static MonoCommands GetWasmFunctionIds(int runtimeId) => new MonoCommands($"getDotnetRuntime({runtimeId}).INTERNAL.mono_wasm_get_func_id_to_name_mappings()");
+        internal JObject ToObject(bool addUrl)
+        {
+            if (!addUrl)
+                return JObject.FromObject(this);
+            JObject obj = JObject.FromObject(this);
```
Comment (ilonatommy) [inline src/mono/browser/debugger/BrowserDebugProxy/DevToolsHelper.cs:331]:
```suggestion
            JObject obj = JObject.FromObject(this);
            if (!addUrl)
                return obj;
```

**Review by ilonatommy — APPROVED** [review]
Is this change testable?
Did this PR merge with main? There's `System.InvalidOperationException : Collection was modified; enumeration operation may not execute.` that was fixed recently

File: src/mono/browser/debugger/BrowserDebugProxy/DevToolsHelper.cs [hunk 2]
```diff
@@ -324,6 +324,14 @@ public static MonoCommands SendDebuggerAgentCommandWithParms(int runtimeId, int
         public static MonoCommands ReleaseObject(int runtimeId, DotnetObjectId objectId) => new MonoCommands($"getDotnetRuntime({runtimeId}).INTERNAL.mono_wasm_release_object('{objectId}')");
 
         public static MonoCommands GetWasmFunctionIds(int runtimeId) => new MonoCommands($"getDotnetRuntime({runtimeId}).INTERNAL.mono_wasm_get_func_id_to_name_mappings()");
+        internal JObject ToObject(bool addUrl)
+        {
+            JObject obj = JObject.FromObject(this);
+            if (!addUrl)
+                return obj;
+            obj[nameof(expression)] = expression + $"//# sourceURL=cdp://dotnet/mono{expression.GetHashCode()}.cdp";
```
Comment (lewing) [inline src/mono/browser/debugger/BrowserDebugProxy/DevToolsHelper.cs:332]:
```suggestion
            obj[nameof(expression)] = $"{expression}//# sourceURL=cdp://dotnet/mono{expression.GetHashCode()}.cdp";
```

Comment (lewing) [inline src/mono/browser/debugger/BrowserDebugProxy/DevToolsHelper.cs:332]:
```suggestion
            if (addUrl)
                obj[nameof(expression)] = $"{expression}//# sourceURL=cdp://dotnet/mono{expression.GetHashCode()}.cdp";
```

**Review by lewing — APPROVED** [review]
I think this is a reasonable compromise

File: src/mono/browser/debugger/BrowserDebugProxy/DevToolsHelper.cs [hunk 3]
```diff
@@ -293,7 +293,10 @@ internal sealed class MonoCommands
         public bool silent { get; set; }
         public bool returnByValue { get; set; } = true;
 
-        public MonoCommands(string expression) => this.expression = expression;
+        public MonoCommands(string expression)
+        {
+            this.expression = $"{expression} //# sourceURL=cdp://debug/eval.cdp";
```
Comment (radical) [inline src/mono/browser/debugger/BrowserDebugProxy/DevToolsHelper.cs:298]: What is picking up `sourceURL` from this comment in the expression?

Comment (thaystg) [reply src/mono/browser/debugger/BrowserDebugProxy/DevToolsHelper.cs:298]: the browser. Then when we receive a `Debugger.scriptParsed` with `url=cdp://debug/eval.cdp` and not empty as it is when we don't pass `cdp://debug/eval.cdp`

**Pull request merged**